oauthenticator/azuread.py

```diff
@@ -69,9 +75,17 @@ async def authenticate(self, handler, data=None):
         resp_json = await self.fetch(req)
 
         access_token = resp_json['access_token']
-
         id_token = resp_json['id_token']
-        decoded = jwt.decode(id_token, options={"verify_signature": False})
+
+        if PYJWT_2:
+            decoded = jwt.decode(
+                id_token,
+                options={"verify_signature": False},
+                audience=self.client_id,
```

Comment:
For anyone else wondering about the addition of `audience` see:
- https://pyjwt.readthedocs.io/en/latest/usage.html
- https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens#payload-claims